### PR TITLE
Update CHANGELOG for 0.8.0 release

### PR DESCRIPTION
**Pull request by rmlarose**
Description
-----------

Checklist
-----------

Check off the following once complete (or if not applicable) after opening the PR. The PR will be reviewed once this checklist is complete and all tests are passing.

- [x] I added unit tests for new code.
- [x] I used [type hints](https://www.python.org/dev/peps/pep-0484/) in function signatures.
- [x] I used [Google-style](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html) docstrings for functions.
- [x] I [updated the documentation](../blob/master/docs/CONTRIBUTING_DOCS.md) where relevant.

If some items remain, you can mark this a [draft pull request](https://github.blog/2019-02-14-introducing-draft-pull-requests/).

Tips
----

- If the validation check fails:

    1. Run `make check-style` (from the root directory
  of the repository) and fix any [flake8](http://flake8.pycqa.org) errors.

    2. Run `make format` to format your code with the [black](https://black.readthedocs.io/en/stable/index.html)
  autoformatter.

  For more information, check the [style guidelines](https://mitiq.readthedocs.io/en/stable/contributing.html#style-guidelines) for Mitiq.
  
- Write "Fixes #XYZ" in the description if this PR fixes Issue #XYZ.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,6 +8,46 @@
 % # " - [Bug Fix]"
 % # " - Fix the bug."
 
+## Version 0.8.0 (May 4th, 2021)
+
+### Summary
+
+This release has the following major components:
+
+- Re-implements local folding functions (`zne.scaling.fold_gates_from_left` and `zne.scaling.fold_gates_from_right`) to make them more uniform at large scale factors and match how they were defined in [Digital zero-noise extrapolation for quantum error mitigation](https://arxiv.org/abs/2005.10921).
```
Comment (andreamari) [inline CHANGELOG.md:17]:
```suggestion
- Re-implements local folding functions (`zne.scaling.fold_gates_from_left`, `zne.scaling.fold_gates_from_right` and `zne.scaling.fold_gates_at_random`) to make them more uniform at large scale factors and match how they were defined in [Digital zero-noise extrapolation for quantum error mitigation](https://arxiv.org/abs/2005.10921).
```

**Review by nathanshammah — APPROVED** [review]
LGTM, thank you.

**Comment by nathanshammah** [comment]
Make sure to update the date to today.

File: CHANGELOG.md [hunk 2]
```diff
@@ -8,6 +8,46 @@
 % # " - [Bug Fix]"
 % # " - Fix the bug."
 
+## Version 0.8.0 (May 4th, 2021)
```
Comment (rmlarose) [inline CHANGELOG.md:11]:
```suggestion
## Version 0.8.0 (May 5th, 2021)
```

**Pull request merged**